plugins/@grouparoo/app-templates/src/source/query/profileProperty.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  ProfilePropertyPluginMethod,
+  plugin,
+  ProfilePropertyPluginMethodResponse,
+} from "@grouparoo/core";
+import {
+  ExecuteQueryMethod,
+  ValidateQueryMethod,
+  queryKey,
+} from "./pluginMethods";
+
+export interface GetProfilePropertyMethod {
+  (argument: {
+    executeQuery: ExecuteQueryMethod;
+    validateQuery?: ValidateQueryMethod;
+  }): ProfilePropertyPluginMethod;
+}
+
+export const getProfileProperty: GetProfilePropertyMethod = ({
+  executeQuery,
+  validateQuery,
+}) => {
+  const profileProperty: ProfilePropertyPluginMethod = async ({
+    connection,
+    appOptions,
+    profile,
+    profilePropertyRule,
+    profilePropertyRuleOptions,
+  }) => {
+    const ruleQuery = profilePropertyRuleOptions[queryKey];
+    const query = await plugin.replaceTemplateProfileVariables(
+      ruleQuery,
+      profile
+    );
+    if (!validateQuery) {
+      validateQuery = validateGenericQuery;
```

Comment:
If a validation query is not provided, I don't think that the generic one should be used without a way to opt-out.  I'd suggest moving this to the arguments so callers can provide `null` if they don't want the default:

```ts
export const getProfileProperty: GetProfilePropertyMethod = ({
  executeQuery,
  validateQuery = validateGenericQuery,
}) => {}

// then to opt-out

getProfileProperty(() => {}, null)
```